### PR TITLE
injections(html): various <script type>

### PR DESCRIPTION
**Pull request by bennypowers**
Replaces the `<script type>` parts from #2577 

image

<details><summary>test file</summary>

```html
<script>
  html`<p></p>`
  html(`<p></p>`)
  svg`<p></p>`
  svg(`<p></p>`)
</script>

<script type="module">
  html`<p></p>`
  html(`<p></p>`)
  svg`<p></p>`
  svg(`<p></p>`)
</script>

<input pattern="[0-9]+">
<input pattern=[0-9]+ type="tel">

<script type="application/json">{ "true": false }</script>
<script type="importmap">{ "true": false }</script>
<script type="text/markdown">## Hello *World*!</script>
<script type="application/graphql">query OK { dokey }</script>
<script type="javascript">function foo () {}</script>
<script type="module">function foo() {}</script>
<script type="application/ecmascript">function foo() {}</script>
<script type="text/ecmascript">function foo() {}</script>
<script type="text/javascript">function foo() {}</script>
```

</details>

### CODE REVIEW

**Review by amaanq — APPROVED** [review]

File: queries/html_tags/injections.scm [hunk 1]
```diff
@@ -64,3 +64,11 @@
...
     ] (#eq? @_attr "pattern")))
 ))
+
+; <script type="${someMimetypeOrConfiguredType}">
+(script_element
+   (start_tag
+      ((attribute
+           (attribute_name) @_attr (#eq? @_attr "type")
+           (quoted_attribute_value (attribute_value) @_type))))
+   (raw_text) @language (#set-lang-from-mimetype! @_type))
```
Comment (lucario387) [inline queries/html_tags/injections.scm:74]: Can you also remove the <script type ...> from some lines above this? This seems to override that already anyways

Comment (lucario387) [reply queries/html_tags/injections.scm:74]: And maybe add some tests :)
Just to be sure

Comment (bennypowers) [reply queries/html_tags/injections.scm]: - removed superfluous query
- explicitly associated "module" script type with javascript grammar
- added test cases to html injections test fixture
- added an ecma test fixture
- added some more descriptive comments to html_tags injections query file
- rebased

Comment (amaanq) [reply queries/html_tags/injections.scm]: Great! I'll merge later today, just hoping Hamsta will maybe see this and go over it...

File: queries/html_tags/injections.scm [hunk 2]
```diff
@@ -27,23 +27,23 @@
...
-  (#eq? @_type "type")
-  (#any-of? @_javascript "text/javascript" "module")
-)
+; <script type="mimetype-or-well-known-script-type">
+(script_element
+   (start_tag
+      ((attribute
+           (attribute_name) @_attr (#eq? @_attr "type")
+           (quoted_attribute_value (attribute_value) @_type))))
+   (raw_text) @language (#set-lang-from-mimetype! @_type))
```
Comment (lucario387) [inline queries/html_tags/injections.scm:36]: Shouldn't this be content instead of language?

Comment (lucario387) [reply queries/html_tags/injections.scm]: This kinda makes me wonder, how was the test still passing before, if `@content` was never assigned to a node?

Comment (bennypowers) [reply queries/html_tags/injections.scm]: i'm guessing it's becaue `metadata.language` is set in the directive

Comment (amaanq) [reply queries/html_tags/injections.scm]: Is this resolved? I'll merge if so

Comment (lucario387) [reply queries/html_tags/injections.scm]: It's resolved as of latest commit

**Comment by amaanq** [comment]
Thanks @bennypowers!

**Pull request merged**